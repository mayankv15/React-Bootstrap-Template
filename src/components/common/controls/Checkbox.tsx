import React from "react";
import {
  Checkbox as MuiCheckbox,
  FormControlLabel,
  FormGroup,
} from "@mui/material";
import {
  CheckBox,
  CheckBoxOutlineBlank,
  CheckCircle,
  RadioButtonUnchecked,
} from "@mui/icons-material";
import Label from "./Label";
import { useTranslation } from "react-i18next";

const Checkbox = (props) => {
  const {
    label,
    color,
    checked,
    value,
    name,
    onChange,
    view,
    inputLabelStyles,
    ...other
  } = props;
  const { t } = useTranslation();

  return (
    <FormGroup>
      <FormControlLabel
        classes={{ label: "MuiFormControlLabel-cust" }}
        control={
          <MuiCheckbox
            onChange={onChange}
            color={color ? color : "primary"}
            checked={checked}
            value={value}
            name={name}
            checkedIcon={view === "circle" ? <CheckCircle /> : <CheckBox />}
            icon={
              view === "circle" ? (
                <RadioButtonUnchecked />
              ) : (
                <CheckBoxOutlineBlank />
              )
            }
          />
        }
        label={<Label title={t(label)} inputLabelStyles={inputLabelStyles} />}
        {...other}
      />
    </FormGroup>
  );
};

export default Checkbox;
