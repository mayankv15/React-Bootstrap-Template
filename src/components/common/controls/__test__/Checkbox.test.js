import React from "react";
import { render } from "@testing-library/react";
import Checkbox from '../Checkbox'

describe("Renders Checkbox", () => {
    it("should render Checkbox correctly", () => {
        const component = render(<Checkbox label="Checkbox Two"
            value='Two' />);
        expect(component).toMatchSnapshot();
    });
});
