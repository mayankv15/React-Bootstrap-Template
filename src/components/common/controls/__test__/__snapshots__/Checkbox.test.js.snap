// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Checkbox should render Checkbox correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-4"
                data-indeterminate="false"
                type="checkbox"
                value="Two"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiFormControlLabel-cust MuiTypography-body1"
          >
            <p
              class="MuiTypography-root makeStyles-label-5 undefined MuiTypography-body1"
            >
              Checkbox Two
               
              
            </p>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-4"
              data-indeterminate="false"
              type="checkbox"
              value="Two"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiFormControlLabel-cust MuiTypography-body1"
        >
          <p
            class="MuiTypography-root makeStyles-label-5 undefined MuiTypography-body1"
          >
            Checkbox Two
             
            
          </p>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
